TYPO3.Media/Classes/TYPO3/Media/Domain/Model/Thumbnail.php

```diff
@@ -126,35 +124,45 @@ protected function setConfiguration(ThumbnailConfiguration $configuration)
      * @param string $value
      * @return mixed
      */
-    protected function getConfigurationValue($value)
+    public function getConfigurationValue($value)
     {
         return Arrays::getValueByPath($this->configuration, $value);
     }
 
     /**
+     * @param Resource $resource
+     * @return void
+     */
+    public function setResource(Resource $resource)
+    {
+        $this->resource = $resource;
+    }
+
+    /**
+     * @param integer $width
+     * @return void
+     */
+    public function setWidth($width)
+    {
+        $this->width = (integer)$width;
+    }
```

Comment:
DimensionsTrait contain only getters, check my comment about the need of the setter for width / height